src/screencast/screencast.ts

```diff
@@ -0,0 +1,201 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { ErrorCodes } from '../common/errorCodes';
+import { encodeMessageForChannel } from '../common/webviewEvents';
+import { ScreencastCDPConnection, vscode } from './cdp';
+import { MouseEventMap, ScreencastInputHandler } from './input';
+
+type NavigationEntry = {
+    id: number;
+    url: string;
+};
+
+export class Screencast {
+    private cdpConnection = new ScreencastCDPConnection();
+    private history: NavigationEntry[] = [];
+    private historyIndex = 0;
+    private inputHandler: ScreencastInputHandler;
+    private backButton: HTMLButtonElement;
+    private forwardButton: HTMLButtonElement;
+    private reloadButton: HTMLButtonElement;
+    private rotateButton: HTMLButtonElement;
+    private urlInput: HTMLInputElement;
+    private screencastImage: HTMLImageElement;
+    private screencastWrapper: HTMLElement;
+    private deviceSelect: HTMLSelectElement;
+    private width = 0;
+    private height = 0;
+
+    constructor() {
+        this.backButton = document.getElementById('back') as HTMLButtonElement;
+        this.forwardButton = document.getElementById('forward') as HTMLButtonElement;
+        this.reloadButton = document.getElementById('reload') as HTMLButtonElement;
+        this.rotateButton = document.getElementById('rotate') as HTMLButtonElement;
+        this.urlInput = document.getElementById('url') as HTMLInputElement;
+        this.screencastImage = document.getElementById('canvas') as HTMLImageElement;
+        this.screencastWrapper = document.getElementById('canvas-wrapper') as HTMLElement;
+        this.deviceSelect = document.getElementById('device') as HTMLSelectElement;
+
+        this.backButton.addEventListener('click', () => this.onBackClick());
+        this.forwardButton.addEventListener('click', () => this.onForwardClick());
+        this.reloadButton.addEventListener('click', () => this.onReloadClick());
+        this.rotateButton.addEventListener('click', () => this.onRotateClick());
+        this.urlInput.addEventListener('keydown', event => this.onUrlKeyDown(event));
+
+        this.deviceSelect.addEventListener('change', () => {
+            if (this.deviceSelect.value.toLowerCase() === 'desktop') {
+                this.width = 0;
+                this.height = 0;
+                this.screencastWrapper.classList.add('desktop');
+            } else {
+                const selectedOption = this.deviceSelect[this.deviceSelect.selectedIndex];
+                const deviceWidth = selectedOption.getAttribute('devicewidth');
+                const deviceHeight = selectedOption.getAttribute('deviceheight');
+                if (deviceWidth && deviceHeight) {
+                    this.width = parseInt(deviceWidth);
+                    this.height = parseInt(deviceHeight);
+                } else {
+                    encodeMessageForChannel(msg => vscode.postMessage(msg, '*'), 'reportError', {
```

Comment:
goes with the above comment, but if you just use 'telemetry' here, you don't need to add reportError and then filter it out in the Panel test